src/urlUtils.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Removes leading and trailing slashes from a route to simplify comparisons
+ * against other paths.
+ *
+ * @external
+ */
+export function normalizeRoute(route: string): string {
+  return stripLeadingSlash(stripTrailingSlash(route));
+}
+
+/**
+ * Removes any leading '/' characters from a string.
+ *
+ * @external
+ */
+export function stripLeadingSlash(str: string): string {
+  return str.replace(/^\/+/, '');
+}
+
+/**
+ * Removes any trailing '/' characters from a string.
+ *
+ * @external
+ */
+export function stripTrailingSlash(str: string): string {
+  return str.replace(/\/+$/, '');
+}
+
+/**
+ * Removes any leading '/' or '#' characters from a string.
```

Comment:
do we have to worry about query strings in urls?

Reply:
This function is for removing the leading / and # of the client route so that client can concat the host url and client route.  As query string is at the tail of the client route, we don't need to worry about it in this case.
And for this function itself, the purpose is just to remove all slash and hashtag. so it should be okay